src/vm/jitinterface.cpp

```diff
@@ -6631,6 +6632,12 @@ const char* CEEInfo::getMethodNameFromMetadata(CORINFO_METHOD_HANDLE ftnHnd, con
         {
             MethodTable* pMT = ftn->GetMethodTable();
             classResult = pMT->GetFullyQualifiedNameInfo(&namespaceResult);
+            // Query enclosingClassName when it is required
+            // The namespace of nested classes is empty
+            if (namespaceResult[0] == '\0' && enclosingClassName != NULL)
```

Comment:
This check should do `pMT->GetClass()->IsNested()`. Not guess based on empty namespace.

Reply:
Thank you, will fix!